### PR TITLE
2 Introduction - v4 Migration Cleanup

### PR DESCRIPTION
**Pull request by rejahrehim**
- [x] This PR handles the issue and requires no additional PRs.
- [x] You have validated the need for this change.

**What did this PR accomplish?**

- Fix linting issues

### CODE REVIEW

File: document/2_Introduction/2_Introduction.md [hunk 1]
```diff
@@ -233,65 +255,63 @@ Of course there are times and circumstances where only one technique is possible
 
 A balanced approach varies depending on many factors, such as the maturity of the testing process and corporate culture. It is recommended that a balanced testing framework should look something like the representations shown in Figure 3 and Figure 4. The following figure shows a typical proportional representation overlaid onto the software development life cycle. In keeping with research and experience, it is essential that companies place a higher emphasis on the early stages of development.
 
-<div align="center">
+![Figure 3: Proportion of Test Effort in SDLC](images/ProportionSDLC.png)
+*Figure 3: Proportion of Test Effort in SDLC*
```
Comment (kingthorin) [inline document/2_Introduction/2_Introduction.md]: I agree the `<br>` was horrible but it's now putting the caption on the same line.
image

Maybe it needs a blank line?

File: document/2_Introduction/2_Introduction.md [hunk 2]
```diff
@@ -233,65 +255,63 @@ Of course there are times and circumstances where only one technique is possible
...
 
-### Deriving Security Test Requirements
------------------------------------
+## Deriving Security Test Requirements
 
 To have a successful testing program, one must know what the testing objectives are. These objectives are specified by the security requirements. This section discusses in detail how to document requirements for security testing by deriving them from applicable standards and regulations, and from positive and negative application requirements. It also discusses how security requirements effectively drive security testing during the SDLC and how security test data can be used to effectively manage software security risks.
 
-#### Testing Objectives
+**Testing Objectives**
```
Comment (kingthorin) [inline document/2_Introduction/2_Introduction.md]: Use level 3s like (Overview, Advantages, Disadvantages) above?

File: document/2_Introduction/2_Introduction.md [hunk 3]
```diff
@@ -366,29 +386,28 @@ Misuse scenarios allow the analysis of the application from the attacker's point
...
+  3. Log in error messages need to be generic
 
 These security requirements need to be documented and tested.
 
-### Security Tests Integrated in Development and Testing Workflows
---------------------------------------------------------------
+## Security Tests Integrated in Development and Testing Workflows
 
-#### Security Testing in the Development Workflow
+**Security Testing in the Development Workflow**
```
Comment (kingthorin) [inline document/2_Introduction/2_Introduction.md]: L3?

File: document/2_Introduction/2_Introduction.md (hunk 2 above)
Comment (kingthorin) [inline document/2_Introduction/2_Introduction.md]: Use level 3s like (Overview, Advantages, Disadvantages) above?

File: document/2_Introduction/2_Introduction.md (hunk 1 above)
Comment (kingthorin) [inline document/2_Introduction/2_Introduction.md]: I agree the `<br>` was horrible but it's now putting the caption on the same line.
image

Maybe they need a blank line?

File: document/2_Introduction/2_Introduction.md (hunk 3 above)
Comment (rejahrehim) [reply document/2_Introduction/2_Introduction.md]: In V4 the TOC doesn't have those sections as headings.

File: document/2_Introduction/2_Introduction.md (hunk 2 above)
Comment (rejahrehim) [reply document/2_Introduction/2_Introduction.md]: If this in L3 then we have to make ### Deriving Functional to L4. How to arrange?

File: document/2_Introduction/2_Introduction.md (hunk 3 above)
Comment (kingthorin) [reply document/2_Introduction/2_Introduction.md]: That's fine isn't it? The ToC can contain whatever level of detail we want.

If you goto the Files tab on your PR then click the little ellipsis (`...`) button and "View File" you'll see how it's going to render. To me the floating bold seems out of place, having them as L3s would make more sense and look better IMHO.

Comment (rejahrehim) [reply document/2_Introduction/2_Introduction.md]: I have updated it with headings, Could you please check it's not breaking the heading hierarchy

File: document/2_Introduction/2_Introduction.md (hunk 1 above)
Comment (rejahrehim) [reply document/2_Introduction/2_Introduction.md]: fixed

File: document/2_Introduction/2_Introduction.md (hunk 2 above)
Comment (kingthorin) [reply document/2_Introduction/2_Introduction.md]: b8c6282 looks good to me.

File: document/2_Introduction/2_Introduction.md (hunk 3 above)
Comment (kingthorin) [reply document/2_Introduction/2_Introduction.md]: Looks good to me. Are you not happy with it?
image

Comment (rejahrehim) [reply document/2_Introduction/2_Introduction.md]: Looks good for me too, Just tried to compare with v4 TOC

Screenshot 2019-07-23 at 5 35 43 PM

Comment (kingthorin) [reply document/2_Introduction/2_Introduction.md]: Ok that's fair, but I think when we re-generate the ToC we can decide how "deep" to include the headings.

Comment (ThunderSon) [reply document/2_Introduction/2_Introduction.md]: I believe these should be L3s. These are subsections of the L2 section.

File: document/2_Introduction/2_Introduction.md [hunk 4]
```diff
@@ -366,29 +386,28 @@ Misuse scenarios allow the analysis of the application from the attacker's point
...
 
 The validation of source code before integration in application builds is usually the responsibility of the senior developer. Such senior developers are also the subject matter experts in software security and their role is to lead the secure code review. They must make decisions on whether to accept the code to be released in the application build or to require further changes and testing. This secure code review workflow can be enforced via formal acceptance as well as a check in a workflow management tool. For example, assuming the typical defect management workflow used for functional bugs, security bugs that have been fixed by a developer can be reported on a defect or change management system. The build master can look at the test results reported by the developers in the tool and grant approvals for checking in the code changes into the application build.
 
-#### Security Testing in the Test Workflow
+**Security Testing in the Test Workflow**
```
Comment (ThunderSon) [inline document/2_Introduction/2_Introduction.md]: This should be L3

File: document/2_Introduction/2_Introduction.md [hunk 5]
```diff
@@ -399,21 +418,21 @@ A testing engineer who validates the security of the application in the integrat
 
 Since these tests are the last resort for fixing vulnerabilities before the application is released to production, it is important that such issues are addressed as recommended by the testing team. The recommendations can include code, design, or configuration change. At this level, security auditors and information security officers discuss the reported security issues and analyze the potential risks according to information risk management procedures. Such procedures might require the development team to fix all high risk vulnerabilities before the application can be deployed, unless such risks are acknowledged and accepted.
 
-#### Developers' Security Tests
+### Developers' Security Tests
 
-##### Security Testing in the Coding Phase: Unit Tests
+**Security Testing in the Coding Phase: Unit Tests**
```
Comment (ThunderSon) [inline document/2_Introduction/2_Introduction.md]: @kingthorin this is up for debate, as this is just explaining what the Security tests are in the coding phase.

File: document/2_Introduction/2_Introduction.md [hunk 6]
```diff
@@ -427,9 +446,9 @@ Threat scenarios for common vulnerabilities that have root causes in insecure co
 
 Source code analysis and unit tests can validate that the code change mitigates the vulnerability exposed by the previously identified coding defect. The results of automated secure code analysis can also be used as automatic check-in gates for version control, for example software artifacts cannot be checked into the build with high or medium severity coding issues.
 
-#### Functional Testers' Security Tests
+### Functional Testers' Security Tests
 
-##### Security Testing During the Integration and Validation Phase: Integrated System Tests and Operation Tests
+**Security Testing During the Integration and Validation Phase: Integrated System Tests and Operation Tests**
```
Comment (ThunderSon) [inline document/2_Introduction/2_Introduction.md]: Similar to the previous comment. Can be up to debate as this is not a subsection, more so giving it its place in the SDLC workflow.

**Comment by ThunderSon** [comment]
I was only looking at one commit, weirdly. All is good.

**Pull request merged**